lti_consumer/lti.py

```diff
@@ -0,0 +1,281 @@
+            response['resultScore'] = round(self.xblock.module_score, 2)
+            response['comment'] = self.xblock.score_comment
+
+        return response
+
+    def delete_result(self, user):  # pylint: disable=unused-argument
+        """
+        Helper request handler for DELETE requests to LTI 2.0 result endpoint
+
+        DELETE handler for lti_2_0_result.  Assumes all authorization has been checked.
+
+        Arguments:
+            request (xblock.django.request.DjangoWebobRequest):  Request object (unused)
+            real_user (django.contrib.auth.models.User):  Actual user linked to anon_id in request path suffix
+
+        Returns:
+            webob.response:  response to this request.  status 200 if success
+        """
+        self.xblock.clear_user_module_score(user)
+        return {}
+
+    def put_result(self, user, result_json):
+        """
+        Helper request handler for PUT requests to LTI 2.0 result endpoint
+
+        PUT handler for lti_2_0_result.  Assumes all authorization has been checked.
+
+        Arguments:
+            request (xblock.django.request.DjangoWebobRequest):  Request object
+            real_user (django.contrib.auth.models.User):  Actual user linked to anon_id in request path suffix
+
+        Returns:
+            webob.response:  response to this request.  status 200 if success.  404 if body of PUT request is malformed
+        """
+        score, comment = parse_result_json(result_json)
+
+        if score is None:
+            # According to http://www.imsglobal.org/lti/ltiv2p0/ltiIMGv2p0.html#_Toc361225514
+            # PUTting a JSON object with no "resultScore" field is equivalent to a DELETE.
+            self.xblock.clear_user_module_score(user)
+        else:
+            self.xblock.set_user_module_score(user, score, self.xblock.max_score(), comment)
+
+        return {}
+
+    def verify_result_headers(self, request, verify_content_type=True):
+        """
+        Helper method to validate LTI 2.0 REST result service HTTP headers.  returns if correct, else raises LtiError
+
+        Arguments:
+            request (xblock.django.request.DjangoWebobRequest):  Request object
+            verify_content_type (bool):  If true, verifies the content type of the request is that spec'ed by LTI 2.0
+
+        Returns:
+            nothing, but will only return if verification succeeds
+
+        Raises:
+            LtiError if verification fails
+        """
+        content_type = request.headers.get('Content-Type')
+        if verify_content_type and content_type != Lti2Consumer.CONTENT_TYPE_RESULT_JSON:
+            log.info("[LTI]: v2.0 result service -- bad Content-Type: %s", content_type)
+            raise LtiError(
+                "For LTI 2.0 result service, Content-Type must be %s.  Got %s",
+                Lti2Consumer.CONTENT_TYPE_RESULT_JSON,
+                content_type
+            )
+
+        _, secret = self.xblock.lti_provider_key_secret
```

Comment:
General nit: I prefer we use `__` to mean 'unused variable' because the `_` semantic gets overloaded w/ the underscore gettext function. It's less of a concern in this xblock but it's a habit I like to follow everywhere.